### ISSUE TITLE
Error on Windows 7

### ISSUE THREAD

**Issue by modulexcite** [coreybutler/porthog#1]
Hi,
Thank you for all the hard work and all the excellent projects you have shared with the community.
I installed porthog (version 0.4) on a Windows 7 SP1 OS but for some reason when checking for certain ports; it's returning the following error:

**porthog 1281**

``` 
child_process.js:508
    throw err;
    ^

Error: Command failed: netstat -aon | findstr :1281 | findstr LISTENING

    at checkExecSyncError (child_process.js:465:13)
    at execSync (child_process.js:505:13)
    at module.exports.stdout.forEach.Object.keys.forEach (C:\Users\mattjs\AppData\Roaming\npm\node_modules\porthog\index.js:15:16)
    at C:\Users\mattjs\AppData\Roaming\npm\node_modules\porthog\cli.js:15:15
    at Array.forEach (native)
    at Object.<anonymous> (C:\Users\mattjs\AppData\Roaming\npm\node_modules\porthog\cli.js:14:6)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12) 
```

However when I run the following command, it seems to produce the desired output
**porthog 80**
```
PORT      PID       PROCESS
--------------------------------------------------
80        -         -
```

Any help would be much appreciated, thank you.

**Comment by coreybutler** [comment]
Hmm. Does `netstat -aon | findstr :1281 | findstr LISTENING` work directly from the command line?

**Comment by modulexcite** [comment]
Thank you for your response, I'm not sure what is going wrong here but here is the output from the both porthog and command line for the same port:

**netstat -aon | findstr :80 | findstr LISTENING**

```
TCP    0.0.0.0:80             0.0.0.0:0              LISTENING       4
TCP    0.0.0.0:8081           0.0.0.0:0              LISTENING       2544
TCP    [::]:80                [::]:0                 LISTENING       4
TCP    [::]:8081              [::]:0                 LISTENING       2544
```


**porthog 80**

```
PORT      PID       PROCESS
--------------------------------------------------
80        -         -

```

and here's the output for a different port:

**netstat -aon | findstr :6060 | findstr LISTENING**
``` not found```

**porthog 6060**

```
child_process.js:508
    throw err;
    ^

Error: Command failed: netstat -aon | findstr :6060 | findstr LISTENING

    at checkExecSyncError (child_process.js:465:13)
    at execSync (child_process.js:505:13)
    at module.exports.stdout.forEach.Object.keys.forEach (C:\Users\mattjs\AppData\Roaming\npm\node_modules\porthog\index.js:15:16)
    at C:\Users\mattjs\AppData\Roaming\npm\node_modules\porthog\cli.js:15:15
    at Array.forEach (native)
    at Object.<anonymous> (C:\Users\mattjs\AppData\Roaming\npm\node_modules\porthog\cli.js:14:6)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
```

**Comment by coreybutler** [comment]
I tracked down and fixed the issue. There was a small uncaught issue with the parser when nothing is running on a specific port. It's been fixed and the updated module has been released on npm as version `0.0.5`.

Thanks for reporting this and helping troubleshoot it!

**Comment by modulexcite** [comment]
Awesome, working again.
Thank you for fixing it, you rock :metal: .
On a different note, shouldn't it also include the results for ports with "Established" status reported by netstat?

**Comment by coreybutler** [comment]
I totally forgot about following-up on this with a comment (sorry). Porthog should not provide results for ports with an `ESTABLISHED` status because these ports are still technically "up for grabs" (i.e. a new web server could be created on such a port). Only ports with a `LISTEN` or `LISTENING` (OS-dependent) status are considered "unavailable" or "hogged by a process". For the purposes of this tool, I want to (for now) keep the scope focused on answering "what's hogging a port I want to use?"

I did some cleanup around this idea and released it as `v0.0.7`.

I see value in easily identifying which local processes are attempting to connect to a specific port, i.e. answer the question of "what process are connecting to my local API?". I'm not sure how many people would need that, but as a desktop developer, I can see the value of something like that. I don't know if that belongs in porthog or a separate tool. I'll have to think about that.

**Comment by modulexcite** [comment]
no worries, makes sense. For some reason, I thought I wouldn't be able to use any ports that already had an established connection. I tried it with a couple of ports to make sure that's not the case.
Thank you for the clarification.
